File: projects/agent_based_modelling/assignment_4/model/road/road.go
package road

import (
	"main/model/road/car"
)

type Road struct {
	length int
	Lane   []*car.Car
}

func New(length int) *Road {
	return &Road{
		length: length,
		Lane:   make([]*car.Car, length),
	}
}

func (r *Road) IsOccupied(n int) bool {
	return r.Lane[n%r.length] != nil
}

File: projects/agent_based_modelling/assignment_4_cpp/Road.cpp
#include "Road.h"


Road::Road(const int length): length(length) {
    lane = new Car*[length];
    for (int i = 0; i < length; i++) {
        lane[i] = nullptr;
    }
}

bool Road::is_occupied(const int n) const {
    return lane[n % length] != nullptr;
}
